lib/createGltf.js

```diff
@@ -251,6 +272,61 @@ function getMaterial(gltf, materials, materialName, options) {
     return materialIndex;
 }
 
+function primitiveInfoMatch(a, b) {
+    return a.hasUvs === b.hasUvs &&
+           a.hasNormals === b.hasNormals;
+}
+
+function splitIncompatibleMaterials(nodes, materials, options) {
+    var splitMaterials = [];
+    var primitiveInfoByMaterial = {};
+    var nodesLength = nodes.length;
+    for (var i = 0; i < nodesLength; ++i) {
+        var meshes = nodes[i].meshes;
+        var meshesLength = meshes.length;
+        for (var j = 0; j < meshesLength; ++j) {
+            var primitives = meshes[j].primitives;
+            var primitivesLength = primitives.length;
+            for (var k = 0; k < primitivesLength; ++k) {
+                var primitive = primitives[k];
+                var hasUvs = primitive.uvs.length > 0;
+                var hasNormals = primitive.normals.length > 0;
+                var primitiveInfo = {
+                    hasUvs : hasUvs,
+                    hasNormals :  hasNormals
+                };
+                var originalMaterialName = defaultValue(primitive.material, 'default');
+                var materialName = originalMaterialName;
+                var suffix = 2;
+                while (defined(primitiveInfoByMaterial[materialName])) {
+                    if (primitiveInfoMatch(primitiveInfo, primitiveInfoByMaterial[materialName])) {
+                        break;
+                    }
+                    materialName = originalMaterialName + '-' + suffix++;
+                }
+
+                primitive.material = materialName;
+                primitiveInfoByMaterial[materialName] = primitiveInfo;
+
+                var material = getMaterialByName(splitMaterials, materialName);
```

Comment:
I timed this just in case. It doesn't look like it's much of a performance concern. Obj parsing just completely trumps it.

Split materials: 5.306ms
Total: 42602.737ms